### DISCUSSION TITLE
trigger_rule related issue

### DISCUSSION THREAD

**Discussion by vishalyadav93** [apache/airflow#27181] [General]
trigger_rule = "all_done" gets task triggered even when upstream task is in  **up_for_retry** state.

**Comment by ephraimbuddy** [comment]
I can't reproduce. What version of airflow are you having this issue?

**Reply by vishalyadav93** [reply]
@ephraimbuddy 

airflow version: 2.2.4

To reproduce:
create task1(say PythonOperator),task2(PythonOperator with trigger rule="all_done")
and raise exception everytime in  task 1 so it goes in up_for_retry state, 

what you will see:
when task 1 goes in **up_for_retry** state, task 2 will get triggered

what it should be:
until task 1 has completed execution (either in success or failed state) then only task 2 should trigger, since trigger_rule for task 2 is "all_done"

**Reply by potiuk** [reply]
@vishalyadav93  If you hava a reproducible DAG and logs showing that - can you please open an issue with all details. I believe you might have made some assumption or mistake - providing easily reproducible example might either show that there is a real error or while preparing it you will find the problem (or we might point to the issue when we see and reproduce it).
